src/main/java/dev/ecattez/shahmat/board/AbstractMovingStrategy.java

```diff
@@ -0,0 +1,19 @@
+package dev.ecattez.shahmat.board;
+
+import dev.ecattez.shahmat.board.move.Movement;
+import dev.ecattez.shahmat.board.move.MovingStrategy;
+import dev.ecattez.shahmat.event.BoardEvent;
+
+import java.util.List;
+import java.util.Optional;
+
+public abstract class AbstractMovingStrategy implements MovingStrategy {
+
+    public Optional<Movement> getMovement(Piece piece, Square from, Square to, Board board, List<BoardEvent> history) {
```

Comment:
Opinionated: Don't you think that the history should be the first argument ? First the context, then the parameters ?

Reply:
I will change the interface to make those changes.